src/codestyle.sh

```diff
@@ -73,6 +73,49 @@ function codestyle_main()
   checkpatch=$(join_path "$kernel_root" 'scripts/checkpatch.pl')
   cmd_script="perl ${checkpatch} ${checkpatch_options}"
 
+  [[ -n "${options_values['START_LINE']}" ]] && start_line=${options_values['START_LINE']}
+  [[ -n "${options_values['END_LINE']}" ]] && end_line=${options_values['END_LINE']}
+
+  if [[ -n "$start_line" || -n "$end_line" ]]; then
+    range_option_flag='true'
+  fi
+
+  # Check if --start-line and --end-line options are being used with a single file
+  if [[ ! -f "$path" && -n "$range_option_flag" ]]; then
+    complain "Invalid path using start-line and end-line option: ${path}"
+    return 2 # ENOENT
+  fi
+
+  # When no --start-line is specified, assign the first line of the file to start_line
+  if [[ -n "$range_option_flag" && -z "$start_line" ]]; then
+    start_line=1
+  fi
+
+  # When no --end-line is specified, assign the last line of the file to end_line
+  if [[ -n "$range_option_flag" && -z "$end_line" ]]; then
+    end_line=$(wc -l < "$FLIST")
+  fi
+
+  # If --start-line and --end-line options are being used, assign a temporary file
+  # to FLIST with the line interval and SPDX License Identifier
+  if [[ -n "$range_option_flag" ]]; then
+    suffix=".${FLIST#*.}"
+    temp_file=$(mktemp --suffix "$suffix" --tmpdir="$PWD")
+
+    if [[ "$start_line" != '1' ]]; then
+      printf '%s\n' '// SPDX-License-Identifier: TEMPFILE' > "$temp_file"
+      ((blank_lines = start_line - 2))
```

Comment:
Is this because of the SPDX line?

Reply:
Yes, the blank lines are added to the temporary file so that checkpatch would show the issue in the right line of the original file. The `start_line - 2` is because e.g. if start_line = 5, it would be necessary to add 4 lines before start_line in the temporary file so that start_line is actually the 5th line in the temp file, and also we have to add the SPDX line as the first line, so we would start the temp file with the SPDX line and 3 blank lines.